src/Ecology.jl

```diff
@@ -271,3 +271,85 @@ function jaccard(asm::EcoBase.AbstractAssemblage)
     hassimilarity(asm) && error("function cannot run with $(typeof(gettypes(asm))) types as contains similarity")
     return jaccard(occurrences(asm))
 end
+
+"""
+generalisedpielou::DiversityLevel, proportions::AbstractArray,
+                       Z::AbstractMatrix)
+generalisedpielou(level::DiversityLevel, proportions::AbstractArray,
+                       sim::AbstractTypes)
+
+Calculates a generalisation of Pielou's evenness for columns
+representing the counts of two subcommunities. Values range from 
+zero to one, with one representing complete evenness within the 
+community. Since this is calculated as H / Hmax, and is just a proportion,
+values remain unchanged regardless of the value(s) of q supplied. 
+
+#### Arguments:
+- `level`: DiversityLevel to calculate at (e.g. subcommunityDiversity)
+
+- `proportions`: population proportions
+
+- `Z`: similarity matrix or
+- `sim`: instance of AbstractTypes
+
+#### Returns:
+- Pielou's evenness metric (at metacommunity level) or metrics (of subcommunities)
+"""
+function generalisedpielou end
+generalisedpielou(level::DiversityLevel,
+                   proportions::AbstractArray,
+                   Z::AbstractMatrix = Matrix(1.0I, size(proportions, 1), size(proportions, 1))) =
+    generalisedpielou(level, proportions, GeneralTypes(Z))
+
+function generalisedpielou(level::DiversityLevel,
+                            proportions::AbstractArray,
+                            sim::AbstractTypes)
+    if (level == subcommunityDiversity)
+        dm = ᾱ
+        ns = vec(sum(x->x>0, proportions, dims=1))
+    elseif (level == metacommunityDiversity)
+        dm = Gamma
+        ns = sum(x->x>0, proportions)
+    else
+        error("Can't calculate richness for $level")
+    end
+    gp = level(dm(Metacommunity(proportions, sim)), 1)
+    gp[!,:diversity] .= log.(gp[!,:diversity])./log.(ns)
+    gp[!,:measure] .= "Pielou"
+    select!(gp, Not(:q))
+    return gp
+end
+
+"""
+    pielou(proportions::AbstractMatrix)
+
+Calculates Pielou's evenness of a series of
+columns representing independent subcommunity counts.
+
+#### Arguments:
+
+- `proportions`: population proportions
+
+#### Returns:
+
+- evenness of subcommunities
+
+#### Example:
+```
+communitymat = [10 20 30 20 0; #5 species (columns) and 6 sites (rows)
+                10 0 50 80 10;
+                60 10 90 0 0; 
+                10 10 10 10 10;
+                70 70 70 70 70;
+                10 0 0 90 0]'
```

Comment:
This is confusing - it would be much better to present this as a dataset with 5 rows and 6 columns if you want to have 5 species and 6 subcommunities.

Reply:
I agree, this was partially based on how it was presented in the docs in ecology.md, line 30. To improve consistency and legibility we might want to consider dropping the adjoint matrix syntax from that line, and making the intended structure of the matrix more apparent. If you generally code your data as columns = sites, rows = spp, then adding something to the start of the docs specifying this would help get people up and running quicker.